storm-webapp/src/main/java/org/apache/storm/daemon/logviewer/webapp/LogviewerResource.java

```diff
@@ -229,9 +229,11 @@ public Response listDumpFiles(@PathParam("topo-id") String topologyId, @PathPara
     @Path("/dumps/{topo-id}/{host-port}/{filename}")
     public Response downloadDumpFile(@PathParam("topo-id") String topologyId, @PathParam("host-port") String hostPort,
                                      @PathParam("filename") String fileName, @Context HttpServletRequest request) throws IOException {
+        String host = request.getRemoteHost();
```

Comment:
It doesn't look right to me to use `getRemoteHost`. 
It returns the client that sends the request. What's needed here is the hostname where the log is.
https://javaee.github.io/javaee-spec/javadocs/javax/servlet/ServletRequest.html#getRemoteHost--

Please correct me if I am wrong

Reply:
fixed